test/mobile/test_bytecode.py

```diff
@@ -15,33 +25,231 @@
+            while current_to_version >= MINIMUM_TO_VERSION:
+                # Backport model to v4 to buffer
+                backport_success = _backport_for_mobile(output_model_path, output_model_path_backport, current_to_version)
+                assert(backport_success)
 
-        expected_result = 3 * torch.ones([2, 4], dtype=torch.float64)
+                backport_version = _get_model_bytecode_version(output_model_path_backport)
+                assert(backport_version == current_to_version)
+
+                buf = io.StringIO()
+                torch.utils.show_pickle.main(
+                    ["", tmpdirname + "/" + output_model_path_backport.name + "@*/bytecode.pkl"],
+                    output_stream=buf)
+                output = buf.getvalue()
+
+                # Load model and run forward method
+                mobile_module = _load_for_lite_interpreter(str(output_model_path))
+                module_input = 1
+                mobile_module_result = mobile_module(module_input)
+                expected_mobile_module_result = 3 * torch.ones([2, 4], dtype=torch.float64)
+                torch.testing.assert_allclose(mobile_module_result, expected_mobile_module_result)
+                current_to_version -= 1
+
+            backport_success = _backport_for_mobile(output_model_path, output_model_path_backport, MINIMUM_TO_VERSION - 1)
+            assert(not backport_success)
+            shutil.rmtree(tmpdirname)
+
+
+    def test_backport_bytecode_from_file_to_file(self):
+        script_module_v5_path = pytorch_test_dri / "cpp" / "jit" / "script_module_v5.ptl"
+
+        with tempfile.TemporaryDirectory() as tmpdirname:
+            backport_model_path = Path(tmpdirname, "backport_script_module_v5.ptl")
+            # backport from file
+            success = _backport_for_mobile(script_module_v5_path, backport_model_path, 4)
+            assert(success)
+
+            buf = io.StringIO()
+            torch.utils.show_pickle.main(["", tmpdirname + "/" + backport_model_path.name + "@*/bytecode.pkl"], output_stream=buf)
+            output = buf.getvalue()
+
+            expected_result = SCRIPT_MODULE_V4_path_BYTECODE_PKL
+            acutal_result_clean = "".join(output.split())
+            expect_result_clean = "".join(expected_result.split())
+            isMatch = fnmatch.fnmatch(acutal_result_clean, expect_result_clean)
+            assert(isMatch)
+
+            # Load model v5 and run forward method
+            mobile_module = _load_for_lite_interpreter(str(backport_model_path))
+            module_input = 1
+            mobile_module_result = mobile_module(module_input)
+            expected_mobile_module_result = 3 * torch.ones([2, 4], dtype=torch.float64)
+            torch.testing.assert_allclose(mobile_module_result, expected_mobile_module_result)
+            shutil.rmtree(tmpdirname)
+
+
+    def test_backport_bytecode_from_file_to_buffer(self):
```

Comment:
What is the goal of this test and the one above? To check just the _backport_for_mobile_xxx mechanism but not the function implementations?

If so then it's fine you pick just a pair of versions to test that rather than all of them.

**However, please use SCRIPT_MODULE_BYTECODE_PKL everywhere in this file rather than directly using the paths or references to things like SCRIPT_MODULE_V4_path_BYTECODE_PKL** Otherwise this is much harder to maintain.

If you want this test to grow to be testing the jumps from all available backport versions then as you did above you should do it in a loop and using SCRIPT_MODULE_BYTECODE_PKL (and add any necessary constant, like script_module_v5_backported_to_v4.ptl, to the SCRIPT_MODULE_BYTECODE_PKL. That should be the only place that hardcodes paths, and other test values for each version.)

Reply:
It's to check just the _backport_for_mobile_xxx mechanism. Main change: 
1. Use `SCRIPT_MODULE_BYTECODE_PKL` here.
2. If it's a temporary export file, prefix with `tmp_*.ptl` to make it clear